### PR TITLE
code-refactoring: Added networking packages

### PR DESCRIPTION
**Pull request by ciiay**
**What type of PR is this?**
/kind chore

**What does this PR do / why we need it**:
Adds the networking package that is responsible for all networking resource generation and CRUD operations on the cluster. 

NOTE: this PR just adds the package files and tests, this code is not invoked at any point yet
**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?
GITOPS-3119: https://issues.redhat.com/browse/GITOPS-3119

**How to test changes / Special notes to the reviewer**:
Unit tests are included.

### CODE REVIEW

**Comment by ciiay** [comment]
Hi @jaideepr97 This PR is ready for review.

File: pkg/networking/ingress.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	"github.com/argoproj-labs/argocd-operator/pkg/mutation"
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrlClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// IngressRequest objects contain all the required information to produce a ingress object in return
+type IngressRequest struct {
```
Comment (jaideepr97) [inline pkg/networking/ingress.go:17]: @ciiay there are a few ingress specific fields that can be included here 
such as `spec.Rules`, `spec.IngressClassname` and `spec.TLS`
see https://github.com/argoproj-labs/argocd-operator/blob/master/controllers/argocd/ingress.go#L123

File: pkg/networking/route.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	"github.com/argoproj-labs/argocd-operator/pkg/mutation"
+	routev1 "github.com/openshift/api/route/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrlClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// RouteRequest objects contain all the required information to produce a route object in return
+type RouteRequest struct {
```
Comment (jaideepr97) [inline pkg/networking/route.go:17]: @ciiay same for routes
please add `spec.Host`, 'spec.Port`, 'spec.Path`, `spec.TLS`, `spec.WildcardPolicy` etc

**Comment by jaideepr97** [comment]
@ciiay please remove non networking package related changes from this PR

**Review by jaideepr97 — APPROVED** [review]
LGTM thanks @ciiay

**Pull request merged**